v2/internal/controllers/generic_controller.go

```diff
@@ -100,7 +101,13 @@ func registerWebhook(mgr ctrl.Manager, obj client.Object) error {
 	return ctrl.NewWebhookManagedBy(mgr).For(obj).Complete()
 }
 
-func RegisterAll(mgr ctrl.Manager, clientFactory ARMClientFactory, objs []registration.StorageType, options Options) error {
+func RegisterAll(
+	mgr ctrl.Manager,
+	clientFactory ARMClientFactory,
+	objs []registration.StorageType,
+	extensionMap map[schema.GroupVersionKind]genruntime.ResourceExtension,
```

Comment:
Suggest simply naming this for what it contains, not the type.
```suggestion
	extensions map[schema.GroupVersionKind]genruntime.ResourceExtension,
```